scripts/update-browser-releases/chrome-version.ts

```diff
@@ -0,0 +1,198 @@
+  let url = `https://chromereleases.googleblog.com/${year}/${month}/${core}_${day}.html`;
+  let releaseNote = await fetch(url);
+
+  if (releaseNote.status == 200) {
+    return url;
+  }
+
+  // Second possibility (less reliable)
+  url = `https://chromereleases.googleblog.com/${year}/${month}/${core}.html`;
+
+  releaseNote = await fetch(url);
+
+  if (releaseNote.status == 200) {
+    return url;
+  }
+  console.warn('Release note not found');
+  return '';
+};
+
+/**
+ * updateChromiumFile - Update the json file listing the browser version of a chromium entry
+ *
+ * @param {object} options The list of options for this type of chromiums.
+ */
+export const updateChromiumFile = async (options) => {
+  //
+  // Get the JSON with the versions from chromestatus
+  //
+  const googleVersions = await fetch(options.chromestatusURL);
+
+  // There is a bug in chromestatus: the first 4 characters are erroneous.
+  // It isn't a valid JSON file.
+  // So we strip these characters and manually parse it.
+  // If one day, the bug is fixed, the next 3 lines can be replaces with:
+  // const versions = await googleVersions.json();
+  let buffer = await googleVersions.text();
+  buffer = buffer.substring(5);
+  const versions = JSON.parse(buffer);
+
+  //
+  // Extract the useful data
+  //
+
+  // Get current stable version
+  const stable = versions[options.releaseBranch].version;
+  const stableReleaseDate = versions[
+    options.releaseBranch
+  ].stable_date.substring(0, 10); // Remove the time part
+
+  // Get current beta version
+  const beta = versions[options.betaBranch].version;
+  const betaReleaseDate = versions[options.betaBranch].stable_date.substring(
+    0,
+    10,
+  ); // Remove the time part
+
+  // Get current nightly (= canary) version
+  const canary = versions[options.nightlyBranch].version;
+  const canaryReleaseDate = versions[
+    options.nightlyBranch
+  ].stable_date.substring(0, 10); // Remove the time part
+
+  //
+  // Get the chrome.json from the local BCD
+  //
+  const file = fs.readFileSync(`${options.bcdFile}`);
+  const chromeBCD = JSON.parse(file.toString());
+
+  //
+  // Update the JSON in memory
+  //
+
+  // Update the stable version entry
+  const releaseNotesURL = await getReleaseNotesURL(
+    stableReleaseDate,
+    options.releaseNoteCore,
+  );
+  if (chromeBCD.browsers[options.bcdBrowserName].releases[stable]) {
+    chromeBCD.browsers[options.bcdBrowserName].releases[stable].release_date =
+      stableReleaseDate;
+    chromeBCD.browsers[options.bcdBrowserName].releases[stable].release_notes =
+      releaseNotesURL;
+    chromeBCD.browsers[options.bcdBrowserName].releases[stable].status =
+      'current';
+  } else {
+    // New entry
+    newBrowserEntry(
+      chromeBCD,
+      options.bcdBrowserName,
+      stable,
+      'current',
+      options.browserEngine,
+      stableReleaseDate,
+      releaseNotesURL,
+    );
+  }
+
+  // Check that all older releases are 'retired'
```

Comment:
https://github.com/mdn/browser-compat-data/blob/f7046ea289b6bee980ee183233675a914a813d23/test/linter/test-browsers-data.ts#L22C1-L34C4

Currently, our tests check if there is only one entry for ['current', 'nightly']. Maybe this code is useful inspiration but what you have here seems to work fine. Just FYI

Reply:
The advantage of my version is that it detects a missing entry, which I prefer in case the script won't run for more than a release cycle: a warning will be issued.